### PR TITLE
added Order, Authentication, User, ShoppingCart controllers

### PR DESCRIPTION
**Pull request by entertainment-veks**
changed their services and daos
added mappers and dtos

### CODE REVIEW

File: src/main/java/hibernate/controller/OrderController.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    public OrderController(OrderService orderService, ShoppingCartService shoppingCartService,
+                           UserService userService, OrderMapper orderMapper) {
+        this.orderService = orderService;
+        this.shoppingCartService = shoppingCartService;
+        this.userService = userService;
+        this.orderMapper = orderMapper;
+    }
+
+    @PostMapping("/complete")
+    public void completeOrder(@RequestBody Long userId) {
```
Comment (Max-Pochepets) [inline src/main/java/hibernate/controller/OrderController.java:33]: It's not the `RequestBody` we are receiving

File: src/main/java/hibernate/controller/OrderController.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        orderService.completeOrder(shoppingCartService.getByUser(userService.get(userId)));
+    }
+
+    @GetMapping
+    public List<OrderResponseDto> getOrdersHistory(@RequestBody Long userId) {
```
Comment (Max-Pochepets) [inline src/main/java/hibernate/controller/OrderController.java:38]: same

File: src/main/java/hibernate/controller/ShoppingCartController.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+                                  MovieSessionService movieSessionService,
+                                  UserService userService) {
+        this.shoppingCartService = shoppingCartService;
+        this.shoppingCartMapper = shoppingCartMapper;
+        this.movieSessionService = movieSessionService;
+        this.userService = userService;
+    }
+
+    @PostMapping("/movie-sessions")
+    public void addMovieSession(@RequestBody Long userId, @RequestBody Long movieSessionId) {
```
Comment (Max-Pochepets) [inline src/main/java/hibernate/controller/ShoppingCartController.java:35]: I guess it is same everywhere

File: src/main/java/hibernate/dao/impl/MovieSessionDaoImpl.java [hunk 4]
```diff
@@ -60,6 +61,18 @@ public MovieSession add(MovieSession movieSession) {
         }
     }
 
+    @Override
+    public Optional<MovieSession> get(Long id) {
+        try (Session session = sessionFactory.openSession()) {
+            Query<MovieSession> query = session.createQuery("from MovieSession"
+                    + " WHERE id = :id", MovieSession.class);
+            query.setParameter("id", id);
```
Comment (Max-Pochepets) [inline src/main/java/hibernate/dao/impl/MovieSessionDaoImpl.java:69]: you could use just `get(ClassType, id)`

File: src/main/java/hibernate/model/dto/OrderResponseDto.java [hunk 5]
```diff
@@ -0,0 +1,42 @@
+package hibernate.model.dto;
+
+import java.util.List;
+
+public class OrderResponseDto {
+    private Long id;
+    private List<Long> tickets;
```
Comment (Max-Pochepets) [inline src/main/java/hibernate/model/dto/OrderResponseDto.java:7]:
```suggestion
    private List<Long> ticketIds;
```

File: src/main/java/hibernate/model/dto/ShoppingCartResponseDto.java [hunk 6]
```diff
@@ -0,0 +1,24 @@
+package hibernate.model.dto;
+
+import java.util.List;
+
+public class ShoppingCartResponseDto {
+    private Long id;
+    private List<Long> tickets;
```
Comment (Max-Pochepets) [inline src/main/java/hibernate/model/dto/ShoppingCartResponseDto.java:7]: same

Comment (Max-Pochepets) [inline src/main/java/hibernate/model/dto/ShoppingCartResponseDto.java:7]: same

File: src/main/java/hibernate/dao/MovieSessionDao.java [hunk 7]
```diff
@@ -3,14 +3,13 @@
...
 import java.util.List;
-import java.util.Optional;
 
 public interface MovieSessionDao {
     List<MovieSession> findAvailableSessions(Long movieId, LocalDate date);
 
     MovieSession add(MovieSession movieSession);
 
-    Optional<MovieSession> get(Long id);
+    MovieSession get(Long id);
```
Comment (Max-Pochepets) [inline src/main/java/hibernate/dao/MovieSessionDao.java:12]: Keep returning Optional from DAO layer

**Review by Max-Pochepets — APPROVED** [review]

File: src/main/java/hibernate/controller/AuthenticationController.java [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/register")
+public class AuthenticationController {
+    private AuthenticationService authenticationService;
```
Comment (KseniiaMakarova) [inline src/main/java/hibernate/controller/AuthenticationController.java:14]:
```suggestion
    private final AuthenticationService authenticationService;
```
let's add `final` to all autowired fields in controllers, services, etc.

File: src/main/java/hibernate/controller/OrderController.java [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+
+@RestController
+@RequestMapping("/orders")
+public class OrderController {
+    private OrderService orderService;
+    private ShoppingCartService shoppingCartService;
+    private UserService userService;
+    private OrderMapper orderMapper;
+
+    public OrderController(OrderService orderService, ShoppingCartService shoppingCartService,
```
Comment (KseniiaMakarova) [inline src/main/java/hibernate/controller/OrderController.java:24]: let's use Autowired annotation above all constructors

File: src/main/java/hibernate/model/dto/OrderResponseDto.java [hunk 10]
```diff
@@ -0,0 +1,42 @@
+package hibernate.model.dto;
+
+import java.util.List;
+
+public class OrderResponseDto {
+    private Long id;
+    private List<Long> ticketIds;
+    private String orderDate;
+    private Long user;
```
Comment (KseniiaMakarova) [inline src/main/java/hibernate/model/dto/OrderResponseDto.java:9]:
```suggestion
    private Long userId;
```

File: src/main/java/hibernate/service/mapper/impl/OrderMapperImpl.java [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+import hibernate.service.mapper.OrderMapper;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.springframework.stereotype.Component;
+
+@Component
+public class OrderMapperImpl implements OrderMapper {
+    @Override
+    public OrderResponseDto parseToDto(Order order) {
+        OrderResponseDto current = new OrderResponseDto();
```
Comment (KseniiaMakarova) [inline src/main/java/hibernate/service/mapper/impl/OrderMapperImpl.java:15]:
```suggestion
        OrderResponseDto dto = new OrderResponseDto();
```

File: src/main/java/hibernate/service/mapper/impl/OrderMapperImpl.java [hunk 12]
```diff
@@ -0,0 +1,24 @@
...
+        current.setId(order.getId());
+        List<Long> tickets = order.getTickets().stream()
+                .map(Ticket::getId)
+                .collect(Collectors.toList());
+        current.setTicketIds(tickets);
+        current.setOrderDate(String.valueOf(order.getOrderDate()));
```
Comment (KseniiaMakarova) [inline src/main/java/hibernate/service/mapper/impl/OrderMapperImpl.java:21]:
```suggestion
        current.setOrderDate(order.getOrderDate().toString());
```
or we can format it

File: src/main/java/hibernate/service/mapper/impl/OrderMapperImpl.java [hunk 13]
```diff
@@ -0,0 +1,24 @@
...
+@Component
+public class OrderMapperImpl implements OrderMapper {
+    @Override
+    public OrderResponseDto parseToDto(Order order) {
+        OrderResponseDto current = new OrderResponseDto();
+        current.setId(order.getId());
+        List<Long> tickets = order.getTickets().stream()
+                .map(Ticket::getId)
+                .collect(Collectors.toList());
+        current.setTicketIds(tickets);
```
Comment (KseniiaMakarova) [inline src/main/java/hibernate/service/mapper/impl/OrderMapperImpl.java:20]: what about userId?

File: src/main/java/hibernate/service/mapper/impl/ShoppingCartMapperImpl.java [hunk 14]
```diff
@@ -0,0 +1,23 @@
...
+import hibernate.service.mapper.ShoppingCartMapper;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.springframework.stereotype.Component;
+
+@Component
+public class ShoppingCartMapperImpl implements ShoppingCartMapper {
+    @Override
+    public ShoppingCartResponseDto parseToDto(ShoppingCart shoppingCart) {
+        ShoppingCartResponseDto current = new ShoppingCartResponseDto();
```
Comment (KseniiaMakarova) [inline src/main/java/hibernate/service/mapper/impl/ShoppingCartMapperImpl.java:15]:
```suggestion
        ShoppingCartResponseDto dto = new ShoppingCartResponseDto();
```

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]
fix travis as well

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**